File: apps/backend/src/main/java/com/supertech/backend/product/service/ProductService.java
package com.supertech.backend.product.service;

import java.util.List;

import com.supertech.backend.product.dto.CreateProductRequest;
import com.supertech.backend.product.dto.ProductResponse;
import com.supertech.backend.product.dto.UpdateProductRequest;

public interface ProductService {
    void create(CreateProductRequest request);

    void update(UpdateProductRequest request, Long id);

    void delete(Long id);

    ProductResponse getById(Long Id);

    List<ProductResponse> getAll();

}

File: apps/frontend/features/product/service/product.service.ts
import { ApiResponse } from "@/common/types/api.types";
import {
  CreateProductRequest,
  ProductResponse,
  UpdateProductRequest,
} from "../types/product.types";
import api from "@/common/lib/axios";

export const getAll = async () => {
  const res = await api.get<ApiResponse<ProductResponse[]>>("/products");
  return res.data;
};

export const getById = async (id: number) => {
  const res = await api.get<ApiResponse<ProductResponse>>(`/products/${id}`);
  return res.data;
};

export const create = async (data: CreateProductRequest) => {
  const res = await api.post<ApiResponse<void>>("/products", data);
  return res.data;
};

export const update = async ({
  id,
  data,
}: {
  id: number;
  data: UpdateProductRequest;
}) => {
  const res = await api.put<ApiResponse<void>>(`/products/${id}`, data);
  return res.data;
};

export const remove = async (id: number) => {
  const res = await api.delete<ApiResponse<void>>(`/products/${id}`);
  return res.data;
};
